src/profile_service.rs

```diff
@@ -104,7 +110,8 @@ fn test_custom_profile() {
     let profile_path = String::from(profile_dir.into_path()
                                     .to_str().unwrap());
 
-    let profile = ProfileService::new(Some(profile_path.to_owned()));
+    let profile = ProfileService::new(ProfilePath::Custom(profile_path
+                                                          .clone()));
```

Comment:
nit: since we're using another line, can we make it more readable? maybe making a variable out of one of those inner values?

```rust
let custom_path = ProfilePath::Custom(profile_path.clone());
let profile = ProfileService::new(cloned_path);
```